src/main/java/co/cask/hydrator/plugin/batch/DynamoDBBatchSource.java

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.hydrator.plugin.batch;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Macro;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.annotation.Plugin;
+import co.cask.cdap.api.data.batch.Input;
+import co.cask.cdap.api.data.batch.InputFormatProvider;
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.dataset.lib.KeyValue;
+import co.cask.cdap.etl.api.Emitter;
+import co.cask.cdap.etl.api.PipelineConfigurer;
+import co.cask.cdap.etl.api.batch.BatchSource;
+import co.cask.cdap.etl.api.batch.BatchSourceContext;
+import co.cask.hydrator.common.ReferencePluginConfig;
+import com.amazonaws.services.dynamodbv2.document.Item;
+import com.google.common.base.Strings;
+import org.apache.hadoop.io.LongWritable;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.annotation.Nullable;
+
+/**
+ * Batch DynamoDB Source Plugin - Reads the data from AWS DynamoDB table.
+ */
+@Plugin(type = BatchSource.PLUGIN_TYPE)
+@Name("DynamoDB")
+@Description("DynamoDB Batch Source that will read the data items from AWS DynamoDB table and emit each item as a " +
+  "JSON string, in the body field of the StructuredRecord, that can be further processed downstream in the pipeline.")
```

Comment:
Agreed.
Once the implementation is completed and code is merged. Will work on this ticket as an enhancement.